### PR TITLE
Add new example files

### PR DESCRIPTION
**Pull request by adamhutchings**

### CODE REVIEW

**Comment by JakeRoggenbuck** [comment]
I really like the struct

File: examples/factorialop.cz [hunk 1]
```diff
@@ -0,0 +1,22 @@
+~ Declares a factorial operator ~
+
+~ Notice the string interpolation apparent.
```
Comment (nathansolomon1678) [inline examples/factorialop.cz]: missing ~ at the end

Comment (adamhutchings) [reply examples/factorialop.cz]: Oof

**Comment by JakeRoggenbuck** [comment]
Maybe the operator would be more difficult to write into the compiler than some basic functionality.
Do you think you could make a new directory called future functionality and place factorialop.cz in there.

**Comment by adamhutchings** [comment]
I'll write that part of the compiler, it doesn't seem too hard.
I assume we'll have string interpolation, and then we just need to check
types, almost the same as a function.
If it really gets too difficult we can scrap it later, but I think it would
be cool and I'll do it myself.
But, of course, you can say sudo get rid of it and I will.

**Comment by JakeRoggenbuck** [comment]
I think you have a good idea of how to do it so... bada bing bada merged!

**Pull request merged**